sycl/source/detail/cg.hpp

```diff
@@ -624,6 +624,10 @@ class CGSemaphoreWait : public CG {
         MExternalSemaphore(ExternalSemaphore), MWaitValue(WaitValue) {}
 
   ur_exp_external_semaphore_handle_t getExternalSemaphore() const {
+    if (MExternalSemaphore == nullptr)
```

Comment:
I think an assertion makes more sense here.

Reply:
I was under the impression that assertions are disabled during release, unless that's not the case here?
https://github.com/intel/llvm/pull/15247#discussion_r1740552157

Reply:
They are, but they still indicate that we don't expect non-nullptr values `MExternalSemaphore` inside this call. I would expect Coverity to pick up on that. If it doesn't, then it's a false positive.

Reply:
I see! Although, would we not want some sort of system to catch if `MExternalSemaphore` is null, even in the release? I also did end up changing the assert in https://github.com/intel/llvm/pull/15247#discussion_r1740552157 into a throw; does it matter if it's not an assert?

Reply:
The intent is different. An exception should be used when something could be false due to invalid input. If something should never be false, it should be an assertion. We shouldn't waste time making this runtime check in release mode since it is (or at least supposed to be) always true. There's little point reporting an exception in this case since the user won't be able to do anything with that information about an internal detail anyway.

Reply:
That makes sense, I've gone ahead and changed it back to an assert here.